### PR TITLE
Add the sla query parameter and include it in the tally response

### PR DESCRIPTION
**Pull request by mstead**
If null is specified for the sla query parameter, we default to a
value of _ANY, which will be set by the snapshot rollers as a
placeholder for the default sla snapshot total.

### CODE REVIEW

File: src/main/resources/liquibase/202002270838-add-sla-column-default-values-and-indexes.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                     unique="false">
+            <column name="sla"/>
+            <column name="product_id" />
+        </createIndex>
+    </changeSet>
+
+    <changeSet id="202002270838-2" author="mstead">
+        <comment>Set the default sla to _ANY</comment>
+        <addDefaultValue tableName="tally_snapshots" columnName="sla" defaultValue="_ANY" />
+    </changeSet>
```
Comment (mstead) [inline src/main/resources/liquibase/202002270838-add-sla-column-default-values-and-indexes.xml]: @kahowell heh... Forgot this wasn't required due to the rule below. I think that the index could be moved over to your snapshot PR to keep it in one changeset file. I can remove this totally from here.

Comment (mstead) [reply src/main/resources/liquibase/202002270838-add-sla-column-default-values-and-indexes.xml]: :+1:  Will remove it.

File: src/main/java/org/candlepin/subscriptions/resource/TallyResource.java [hunk 2]
```diff
@@ -121,4 +123,11 @@ public TallyReport getTallyReport(String productId, @NotNull String granularity,
         return report;
     }
 
+    private String getSnapshotQueryValue(String originalValue) {
```
[the remark this answers is not in the document]
Comment (mstead) [reply src/main/java/org/candlepin/subscriptions/resource/TallyResource.java]: :+1:  Yep. Had that on my mind yesterday. Was waiting to get your PR merged first. I'll make sure I do this.

File: src/main/java/org/candlepin/subscriptions/db/TallySnapshotRepository.java [hunk 3]
```diff
@@ -38,10 +38,11 @@
...
  */
 public interface TallySnapshotRepository extends JpaRepository<TallySnapshot, UUID> {
-    Page<TallySnapshot>
-        findByAccountNumberAndProductIdAndGranularityAndSnapshotDateBetweenOrderBySnapshotDate(
-        String accountNumber, String productId, Granularity granularity, OffsetDateTime beginning,
-        OffsetDateTime ending, Pageable pageable);
+
+    @SuppressWarnings("linelength")
+    Page<TallySnapshot> findByAccountNumberAndProductIdAndGranularityAndServiceLevelAndSnapshotDateBetweenOrderBySnapshotDate(
+        String accountNumber, String productId, Granularity granularity, String serviceLevel,
```
[the remark this answers is not in the document]
Comment (mstead) [reply src/main/java/org/candlepin/subscriptions/db/TallySnapshotRepository.java]: :+1:

File: src/main/java/org/candlepin/subscriptions/resource/TallyResource.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mstead) [reply src/main/java/org/candlepin/subscriptions/resource/TallyResource.java]: DONE

File: src/main/java/org/candlepin/subscriptions/db/TallySnapshotRepository.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mstead) [reply src/main/java/org/candlepin/subscriptions/db/TallySnapshotRepository.java]: DONE

File: src/main/resources/liquibase/202002270838-add-sla-column-default-values-and-indexes.xml (hunk 1 above)
Comment (mstead) [reply src/main/resources/liquibase/202002270838-add-sla-column-default-values-and-indexes.xml]: DONE

**Comment by mstead** [comment]
> Changelog needs to be dropped, consider SLA enum, other than that I'm happy with this.

DONE

**Comment by mstead** [comment]
NOT READY FOR MERGE.

**Comment by mstead** [comment]
Ok, this is now ready for review.

Take note that with these changes we are storing the sla value in the DB as the enum string value (SELF_SUPPORT), but the 'defined' value will be used in the API response (Self-Support).

I'd like some feedback on this since it is a bit of a change.

**Comment by kahowell** [comment]
> Ok, this is now ready for review.
> 
> Take note that with these changes we are storing the sla value in the DB as the enum string value (SELF_SUPPORT), but the 'defined' value will be used in the API response (Self-Support).
> 
> I'd like some feedback on this since it is a bit of a change.

I'm not a fan of the DB value changes (but could be convinced), but other than that the changes look good.

File: src/main/java/org/candlepin/subscriptions/db/TallySnapshotRepository.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mstead) [reply src/main/java/org/candlepin/subscriptions/db/TallySnapshotRepository.java]: Using the enum here seemed to force us to store the enum name so I decided not to do this.

**Pull request merged**